src/app/clusters/thermostat-server/thermostat-server.cpp

```diff
@@ -105,43 +104,40 @@ MatterThermostatClusterServerPreAttributeChangedCallback(const app::ConcreteAttr
     int16_t AbsMaxCoolSetpointLimit;
     int16_t MinCoolSetpointLimit;
     int16_t MaxCoolSetpointLimit;
-    int8_t DeadBand         = 0;
-    int16_t DeadBandTemp    = 0;
-    uint32_t FeatureMap     = 0;
-    bool AutoSupported      = false;
-    bool HeatSupported      = false;
-    bool CoolSupported      = false;
-    bool OccupancySupported = false;
+    int8_t DeadBand      = 0;
+    int16_t DeadBandTemp = 0;
     int16_t OccupiedCoolingSetpoint;
     int16_t OccupiedHeatingSetpoint;
     int16_t UnoccupiedCoolingSetpoint;
     int16_t UnoccupiedHeatingSetpoint;
+    uint32_t OurFeatureMap;
+    bool AutoSupported      = false;
+    bool HeatSupported      = false;
+    bool CoolSupported      = false;
+    bool OccupancySupported = false;
 
-#ifndef FEATURE_MAP_OVERIDE
-    emberAfReadServerAttribute(endpoint, chip::app::Clusters::Thermostat::Id,
-                               chip::app::Clusters::Thermostat::Attributes::FeatureMap::Id,
-                               reinterpret_cast<uint8_t *>(&FeatureMap), sizeof(FeatureMap));
-#else
-    FeatureMap = FEATURE_MAP_OVERIDE;
-#endif
+    if (FeatureMap::Get(endpoint, &OurFeatureMap) != EMBER_ZCL_STATUS_SUCCESS)
+        OurFeatureMap = FEATURE_MAP_DEFAULT;
 
-    if (FeatureMap & 1 << 5) // Bit 5 is Auto Mode supported
-    {
+    if (OurFeatureMap & 1 << 5) // Bit 5 is Auto Mode supported
         AutoSupported = true;
-        if (MinSetpointDeadBand::Get(endpoint, &DeadBand) != EMBER_ZCL_STATUS_SUCCESS)
-        {
-            DeadBand = kDefaultDeadBand;
-        }
-        DeadBandTemp = static_cast<int16_t>(DeadBand * 10);
-    }
-    if (FeatureMap & 1 << 0)
+
+    if (OurFeatureMap & 1 << 0)
         HeatSupported = true;
 
-    if (FeatureMap & 1 << 1)
+    if (OurFeatureMap & 1 << 1)
         CoolSupported = true;
 
-    if (FeatureMap & 1 << 2)
-        OccupancySupported = true;
+    if (OurFeatureMap & 1 << 2)
```

Comment:
Previous code I could read "OccupancySupported = true;" to figure out what this bit 2 is.

new code is totally opaque. At least a comment should be here or ideally constants that are readable.